### ISSUE TITLE
Resource artrifactory_ldap_setting forcing user_dn_pattern

### ISSUE THREAD

**Issue by jcoste-orange** [jfrog/terraform-provider-artifactory#355]
**Describe the bug**
I try to define a LDAP setting like the one I configure in the UI.
But the provider forces us to give a user_dn_pattern which is not the case in the UI.

With no user_dn_pattern or with an empty string I've got an error.

```
resource "artifactory_ldap_setting" "ldap_server" {
  key = "ldap_server"
  enabled = true
  ldap_url = "ldap://ldapserver:389/dc=mycorp,dc=com"
  user_dn_pattern = ""
  email_attribute = "mail"
  auto_create_user = false
  ldap_poisoning_protection = true
  allow_user_to_access_profile = false
  paging_support_enabled = true
  search_filter = "(cn={0})"
  search_base = ""
  search_sub_tree = true
  manager_dn = "cn=artifactory,dc=mycorp,dc=com"
  manager_password = "xxx"
}
```
Error :
```
$ terraform plan
╷
│ Error: expected "user_dn_pattern" to not be an empty string, got 
│ 
│   with artifactory_ldap_setting.ldap_server,
│   on ldap.tf line 5, in resource "artifactory_ldap_setting" "ldap_server":
│    5:   user_dn_pattern = ""
│ 
```
Artifactory version : 7.33.12
Artifactory TF Provider version : 2.22.0
Terraform version : 1.0.5

The fragment of system configiration for this ldap is :
```
<ldapSetting>
  <key>ldap_server</key>
  <enabled>true</enabled>
  <ldapUrl>ldap://ldapserver:389/dc=mycorp,dc=com</ldapUrl>
  <userDnPattern/>
  <search>
    <searchFilter>cn={0}</searchFilter>
    <searchBase/>
    <searchSubTree>true</searchSubTree>
    <managerDn>cn=artifactory,dc=mycorp,dc=com</managerDn>
    <managerPassword>xxxx</managerPassword>
  </search>
  <autoCreateUser>false</autoCreateUser>
  <emailAttribute>mail</emailAttribute>
  <ldapPoisoningProtection>true</ldapPoisoningProtection>
  <allowUserToAccessProfile>false</allowUserToAccessProfile>
  <pagingSupportEnabled>true</pagingSupportEnabled>
</ldapSetting>
```

**Requirements for and issue**
- [X] A fully functioning terraform snippet that can be copy&pasted (no outside files or ENV vars unless that's part of the issue)
- [X] Your version of artifactory (you can `curl` it at `$host/artifactory/api/system/version`
- [X] Your version of terraform

**Expected behavior**
Have the ability to not give a user DN Pattern like in the UI

**Additional context**

Thanks

**Comment by maheshjfrog** [comment]
hi @jcoste-orange ... Does providing "" ie: empty string input for "user_dn_pattern" field makes sense?

**Comment by maheshjfrog** [comment]
I have a PR which makes the "user_dn_pattern" field optional. 
Provider currently enforces non-empty input and a valid ldap string.

**Comment by maheshjfrog** [comment]
Hi @jcoste-orange  ... I have a fix in now in master(version 2.22.2). 
Details: Made "user_dn_pattern" field optional. 
For a valid "artifactory_ldap_setting" resource, please specify "user_dn_pattern" and/or "search_filter" with non-empty and valid value. 
Note: empty value to "user_dn_pattern" and "search_filter" will fail validation.

**Comment by jcoste-orange** [comment]
Awesome, it works like a charm. Thanks.
Yes I known that both fields "user_dn_pattern" and "search_filter" can't be empty.
